src/fields2cover/path_planning/path_planning.cpp

```diff
@@ -125,11 +120,23 @@ std::vector<std::pair<F2CPoint, double>> PathPlanning::simplifyConnection(
     }
   }
 
+  if (p1.distance(p2) < 6.0 * safe_dist || vp.size() <2) {
+    path.emplace_back(p2, ang2);
+    return path;
+  }
+
   for (int i = 1; i < vp.size() - 1; ++i) {
     double dist_in  = vp[i].distance(vp[i-1]);
     double dist_out  = vp[i].distance(vp[i+1]);
     double d_in  = min(0.5 * dist_in,  safe_dist);
     double d_out  = min(0.5 * dist_out,  safe_dist);
+    // I haven't checked if this is possible, but
```

Comment:
We were assuming that those points are not duplicated. Maybe a huge assumption. Can you add a little filter?